core/plugins/org.polarsys.capella.core.platform.sirius.sirius.validation/src/org/polarsys/capella/core/platform/sirius/sirius/validation/ddiagram/DescriptionLinkModifierHandler.java

```diff
@@ -0,0 +1,230 @@
+	}
+	
+	public static void addElementToDescription(String qName, Attributes attributes, StringBuilder description) {
+	    // '<'
+		description.append(IConstantValidation.LESS_THAN);
+	    // element name
+		description.append(qName);
+	    if (attributes.getLength() == 0) {
+	      // close tab ">" if no attributes
+	      // empty line correction in description after publication
+	    	description.append(IConstantValidation.GREATER_THAN);
+	    } else {
+	      // fore all attributes
+	      for (int i = 0; i < attributes.getLength(); i++) {
+	        // above filter state the image source (which could be relative or absolute path)
+	        String attValue = attributes.getValue(i);
+	        String attName = attributes.getQName(i);
+	        // add other attribute
+	        description.append(IConstantValidation.SPACE);
+	        description.append(attName);
+	        description.append(IConstantValidation.EQUAL);
+	        description.append(IConstantValidation.DOUBLE_QUOTES);
+	        description.append(attValue);
+	        description.append(IConstantValidation.DOUBLE_QUOTES);
+	      }
+	      if (qName.equals("img")) {
+	    	  description.append("/");
+	      }
+	      // close start Element
+	      description.append(IConstantValidation.GREATER_THAN);
+	    }
+	}
+	
+	private class LinkParserHandler extends DefaultHandler {
+		private EObject currentTargetElement;
+		private String currentLinkHref;
+		private Attributes currentLinkAttributes;
+        // value calculated
+        private boolean valueToAdd = false;
+        // merge the value
+        private StringBuilder elementValue = new StringBuilder();
+        //in case the a tag has other tags inside 
+        private boolean insideLink = false;
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public void characters(char ch[], int start, int length) throws SAXException {
+          valueToAdd = true;
+          // value of an element
+          elementValue.append(new String(ch, start, length));
+        }
+
+        @Override
+        public void startElement(String uri, String localName, String qName, Attributes attributes)
+            throws SAXException {
+          // startElement can be called just after value is calculated for nested element
+          // example: <p> value1 <a href="hvalue"> value2 </a></p>
+          // so need to add the value to the result before starting new element
+          if (valueToAdd) {
+        	  updatedDescription.append(elementValue.toString());
+            valueToAdd = false;
+          }
+          elementValue = new StringBuilder(0);
+
+          // if qName is break
+          // an endElement will add the tag value to the result
+          if (qName.equals(IConstantValidation.XHTML_BREAK_ELEMENT)) {
+            return;
+          }
+
+          // if not a tag, add avery thing to result
+          if(qName.equalsIgnoreCase(IConstantValidation.XHTML_A_TAG)) {
```

Comment:
Sill unformatted code. Before committing you can select all classes and apply the code formatter once for all.